selfhost/parser.jakt

```diff
@@ -155,6 +170,164 @@ boxed enum ParsedStatement {
     Try(stmt: ParsedStatement, error_name: String, error_span: Span, catch_block: ParsedBlock, span: Span)
     Garbage(Span)
 
+    function equals(this, anon parsed_statement: ParsedStatement) -> bool => match this {
```

Comment:
Can we put the equality logic in a separate file? Probably better separate so that the parser file is focused on the parser.

Reply:
This may mean moving the definitions to a different file, or using a function instead of a method.

Reply:
That's probably for the best. Will change it to a different file.

Reply:
I remove all the .equals methods into separate functions. For now, this functions live on the parser file since it's not possible to put them in a separate file due to imports not handling circular references.